shared/protocol/src/main/java/io/pravega/shared/NameUtils.java

```diff
@@ -624,7 +638,7 @@ public static String getStreamForReaderGroup(String groupNameName) {
      */
     public static String validateUserStreamName(String name) {
         Preconditions.checkNotNull(name);
-        Preconditions.checkArgument(name.length() < 256, "Name must have less than 256 characters");
+        Preconditions.checkArgument(name.length() < MAX_GIVEN_NAME_SIZE + 1, String.format("Name cannot exceed %d characters", MAX_GIVEN_NAME_SIZE));
```

Comment:
Use `<=` to avoid having to do arithmetic every time this executes.
Do not use String.format. If you check the syntax of `Preconditions.checkArgument`, you'll see that it already takes a formatted string as arg.